tests/test_construction_missing_annotations.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import annotations
+
+from typing import ForwardRef, Optional
+
+import pytest
+
+from pydantic.main import BaseModel
+
+
+def test_missing_annotation_warning_defaults_to_true():
+
+    with pytest.raises(UserWarning, match='is not fully defined'):
+
+        Bar = ForwardRef('Bar')
+
+        class Foo(BaseModel):
+            a: Bar
+
+        class Bar(BaseModel):
+            foo: Optional[Foo] = None
```

Comment:
Which line exactly is expected to raise the error? Is it line 17? Or line 20? (Perhaps indicate so in the docstring of the test.)